Modules/_ctypes/callbacks.c

```diff
@@ -146,47 +146,55 @@ static void _CallPythonObject(void *mem,
                               int flags,
                               void **pArgs)
 {
-    Py_ssize_t i;
-    PyObject *result;
-    PyObject *arglist = NULL;
-    Py_ssize_t nArgs;
+    PyObject *result = NULL;
+    PyObject **args = NULL;
+    Py_ssize_t i = 0, j = 0, n_args = 0;
     PyObject *error_object = NULL;
     int *space;
     PyGILState_STATE state = PyGILState_Ensure();
 
-    nArgs = PySequence_Length(converters);
+    n_args = PySequence_Length(converters);
```

Comment:
Converters are always tuple: you can add ``assert(PyTuple_Check(converters));`` at the function entry and then use PyTuple_GET_SIZE and PyTuple_GET_ITEM.